### PR TITLE
Clean up forecast modules

### PR DESCRIPTION
**Pull request by wenfeiy-db**
* Use common `generate_cutoffs` for pmdarima and prophet training
* Move import outside of prophet estimator and fit function
* Improve test coverage

Manually tested e2e automl forecasting with custom wheel
* high_temp: https://e2-dogfood.staging.cloud.databricks.com/?o=6051921418418893#notebook/4240927549507925/command/4240927549507935
* store_item_demand (multivariate): https://e2-dogfood.staging.cloud.databricks.com/?o=6051921418418893#notebook/4240927549508282/command/4240927549508293
* weekly_count (weekly): https://e2-dogfood.staging.cloud.databricks.com/?o=6051921418418893#notebook/4240927549509244/command/4240927549509254

### CODE REVIEW

File: runtime/tests/automl_runtime/forecast/prophet/model_test.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ def test_model_save_and_load(self):
         # Check the prediction with the saved model
         prophet_model.predict(self.X)
         forecast_pd = prophet_model._model_impl.python_model.predict_timeseries()
+        print(forecast_pd)
```
Comment (lu-wang-dl) [inline runtime/tests/automl_runtime/forecast/prophet/model_test.py:62]: remove this print.

File: runtime/tests/automl_runtime/forecast/utils_test.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+
+class TestGenerateCutoffs(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.X = pd.DataFrame(
+            pd.date_range(start="2020-07-01", end="2020-08-30", freq='d'), columns=["ds"]
+        ).rename_axis("y").reset_index()
+
+    def test_generate_cutoffs_success(self):
```
Comment (lu-wang-dl) [inline runtime/tests/automl_runtime/forecast/utils_test.py:31]: Do we also want to test the failed conditions for `generate_cutoffs`?

**Review by lu-wang-dl — APPROVED** [review]
Also double check on the test coverage.

**Comment by wenfeiy-db** [comment]
> Also double check on the test coverage.

I think the test coverage looks good.

---------- coverage: platform linux, python 3.8.12-final-0 -----------
Name                                                             Stmts   Miss  Cover
------------------------------------------------------------------------------------
databricks/__init__.py                                               1      0   100%
databricks/automl_runtime/__init__.py                                0      0   100%
databricks/automl_runtime/forecast/__init__.py                       1      0   100%
databricks/automl_runtime/forecast/pmdarima/__init__.py              0      0   100%
databricks/automl_runtime/forecast/pmdarima/diagnostics.py          26      0   100%
databricks/automl_runtime/forecast/pmdarima/model.py               129      1    99%
databricks/automl_runtime/forecast/pmdarima/training.py             61      1    98%
databricks/automl_runtime/forecast/pmdarima/utils.py                24      0   100%
databricks/automl_runtime/forecast/prophet/__init__.py               0      0   100%
databricks/automl_runtime/forecast/prophet/forecast.py              67      1    99%
databricks/automl_runtime/forecast/prophet/model.py                 98      3    97%
databricks/automl_runtime/forecast/utils.py                         23      3    87%
databricks/automl_runtime/hyperopt/__init__.py                       0      0   100%
databricks/automl_runtime/hyperopt/early_stop.py                    25      0   100%
databricks/automl_runtime/sklearn/__init__.py                        3      0   100%
databricks/automl_runtime/sklearn/base_datetime_transformer.py      30      0   100%
databricks/automl_runtime/sklearn/column_selector.py                14      0   100%
databricks/automl_runtime/sklearn/date_transformer.py                7      0   100%
databricks/automl_runtime/sklearn/timestamp_transformer.py           8      0   100%
databricks/automl_runtime/utils.py                                  10      0   100%
databricks/automl_runtime/version.py                                 0      0   100%
------------------------------------------------------------------------------------
TOTAL                                                              527      9    98%

File: runtime/tests/automl_runtime/forecast/utils_test.py (hunk 2 above)
Comment (wenfeiy-db) [reply runtime/tests/automl_runtime/forecast/utils_test.py:31]: We have the two failure cases below - `test_generate_cutoffs_failure_horizon_too_large` and `test_generate_cutoffs_less_data`. Is there other failure cases to cover?

**Pull request merged**